app/com/lynxanalytics/biggraph/frontend_operations/VertexAttributeOperations.scala

```diff
@@ -7,17 +7,87 @@ import com.lynxanalytics.biggraph.graph_api.Scripting._
+  def projectIdSet: VertexSet
+  def tableAttributes = table.vertexAttributes
+  val projectAttrNames = projectAttributes.iterator.map(_._1).toList.sorted
+
+  params ++= List(
+    Choice("id_attr", "ID attribute",
+      options = FEOption.unset +: FEOption.list(projectAttrNames)),
+    Choice("id_column", "ID column",
+      options = FEOption.unset +: table.vertexAttrList),
+    Param("prefix", "Name prefix for the imported attributes"),
+    Choice("unique_keys", "Assert unique ID attribute values", options = FEOption.boolsDefaultFalse),
+    Choice("if_exists", "What happens if an attribute already exists",
+      options = FEOption.list(
+        "Overwrite from the table", "Keep the graph's version", "Use the table's version",
+        "They must match", "Disallow this")))
+  def enabled =
+    project.hasVertexSet &&
+      FEStatus.assert(projectAttrNames.nonEmpty, "No attributes to use as key.")
+
+  def apply() = {
+    val idAttrName = params("id_attr")
+    val idColumnName = params("id_column")
+    assert(idAttrName != FEOption.unset.id, "The ID attribute parameter must be set.")
+    assert(idColumnName != FEOption.unset.id, "The ID column parameter must be set.")
+    val idAttr = projectAttributes(idAttrName).asString
+    val idColumn = tableAttributes(idColumnName).asString
+    val uniqueKeys = params("unique_keys").toBoolean
+    val edges = if (uniqueKeys) {
+      val op = graph_operations.EdgesFromUniqueBipartiteAttributeMatches()
+      op(op.fromAttr, idAttr)(op.toAttr, idColumn).result.edges
+    } else {
+      val op = graph_operations.EdgesFromLookupAttributeMatches()
+      op(op.fromAttr, idAttr)(op.toAttr, idColumn).result.edges
+    }
+    val prefix = if (params("prefix").nonEmpty) params("prefix") + "_" else ""
+    for ((name, attr) <- tableAttributes) {
+      val tableAttr = attr.pullVia(edges)
+      if (projectAttrNames.contains(prefix + name)) {
+        val graphAttr = projectAttributes(prefix + name)
+        params("if_exists") match {
+          case _ if prefix + name == idAttrName => // Leave the key alone.
+          case "Disallow this" => throw new AssertionError(
+            s"Cannot import column `${prefix + name}`. Attribute already exists.")
+          case "Overwrite from the table" =>
+            assert(
+              tableAttr.typeTag.tpe =:= graphAttr.typeTag.tpe,
+              "$prefix$name in the graph has a different type than in the table.")
+            projectAttributes(prefix + name) = ops.unifyAttribute(tableAttr, graphAttr)
+          case "Keep the graph's version" =>
+          case "Use the table's version" =>
+            projectAttributes(prefix + name) = tableAttr
+          case "They must match" =>
+            projectAttributes(prefix + name) =
+              graph_operations.DeriveScala.deriveAndInferReturnType(
+                """
+                assert(t.isEmpty || t == g, s"ATTR does not match on ${id.get}: $t <> $g")
```

Comment:
`|| g.isEmpty` before the `==`?

Reply:
> `|| g.isEmpty` before the `==`?

It was intentionally like this, but I agree that it was a bad intention. 😄  Fixed, thanks.